pkg/controller/keyvault/event_mirror.go

```diff
@@ -24,8 +25,19 @@ import (
 	"github.com/Azure/aks-app-routing-operator/pkg/util"
 )
 
-var (
-	eventMirrorControllerName = controllername.New("keyvault", "event", "mirror")
+var eventMirrorControllerName = controllername.New("keyvault", "event", "mirror")
+
+const (
+	involvedObjectKindField = "involvedObject.kind"
+	eventReasonFailedMount  = "FailedMount"
+	reasonField             = "reason"
+	eventKindPod            = "Pod"
+)
+
+// EventMirrorSelector is a selector for Events that are relevant to the EventMirror controller
+var EventMirrorSelector = fields.AndSelectors(
```

Comment:
more of a high level question that may not be relevant to this PR but why don't we also mirror events from SPC? isn't that what contains the aggregation for events from CSI?

Reply:
I don't know. Would maybe make sense to mirror from SPC as well if there's useful info reflected there. Original owners of this probably determined that the Pod FailedMount error is most relevant though

We should probably mirror events to Gateway and NginxIngressController (in case of default certs) as well.